docker/app.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from flask import Flask, request, jsonify
+import os
+import socket
+import tempfile
+import numpy as np
+from LASER.source.lib.text_processing import BPEfastApply
+from LASER.source.embed import *
+
+app = Flask(__name__)
+app.config['JSON_AS_ASCII'] = False
+
+
+@app.route("/")
+def root():
+    print("/")
+    html = "<h3>Hello {name}!</h3>" \
+           "<b>Hostname:</b> {hostname}<br/>"
+    return html.format(name=os.getenv("LASER", "world"), hostname=socket.gethostname())
+
+
+@app.route("/vectorize")
+def vectorize():
+    content = request.args.get('q')
+    lang = request.args.get('lang')
+    embedding = ''
+    if lang == None or len(lang) == 0:
+        lang = "en"
+    # encoder
+    model_dir = os.path.dirname(os.path.realpath(__file__))
+    model_dir = os.path.join(model_dir, "LASER")
+    model_dir = os.path.join(model_dir, "models")
+    encoder_path = os.path.join(model_dir, "bilstm.93langs.2018-12-26.pt")
+    bpe_codes_path = os.path.join(model_dir, "93langs.fcodes")
+    print(' - Encoder: loading {}'.format(encoder_path))
+    encoder = SentenceEncoder(encoder_path,
+                              max_sentences=None,
+                              max_tokens=12000,
+                              sort_kind='mergesort',
+                              cpu=True)
+    with tempfile.TemporaryDirectory() as tmpdir:
+        ifname = content  # stdin will be used
+        bpe_fname = os.path.join(tmpdir, 'bpe')
```

Comment:
This part of code is copied from embed.py
Why have you omitted Tokenization ?
This is crucial to get correct results, in particular for Greek (which is romanized), Chinese and Japanese which are segmented, but also for all other languages since punctuations, apostrophes have to be correctly handled.
Tokenization also include some normalizations and lower-casing of all texts !

Reply:
I wanted to have a first working application for simple alphabets then afterwards wanted to recover the full set of options you defined in the embed.py

As you've putted this code inside of the __main__ function, I'm not able to use it as an external module.

Reply:
You can call the tokenizer fro ma function, you just need lines similar to:
            Token(ifname,                                                                                                                    
                  tok_fname,                                                                                                                 
                  lang=args.token_lang,                                                                                                      
                  romanize=True if args.token_lang == 'el' else False,                                                                       
                  lower_case=True, gzip=False,                                                                                               
                  verbose=args.verbose, over_write=False)                                                                                    
            ifname = tok_fname                                                                                                               
before BPEfastApply()

Reply:
Ok I'll try to put it then